src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNINpHandle.cs

```diff
@@ -213,7 +211,7 @@ public override uint Send(SNIPacket packet)
             }
         }
 
-        public override uint SendAsync(SNIPacket packet, SNIAsyncCallback callback = null)
+        public override uint SendAsync(SNIPacket packet, bool disposePacketAfterSendAsync, SNIAsyncCallback callback = null)
         {
             SNIAsyncCallback cb = callback ?? _sendCallback;
             packet.WriteToStreamAsync(_stream, cb, SNIProviders.NP_PROV);
```

Comment:
Pass the disposePacketAfterSendAsync to the packet.WriteToStreamAsync(... ) in SNINpHandle.cs